gpt_index/langchain_helpers/chain_wrapper.py

```diff
@@ -93,7 +93,12 @@ def _predict(self, prompt: Prompt, **prompt_args: Any) -> str:
         if self.retry_on_throttling:
             llm_prediction = retry_on_exceptions_with_backoff(
                 lambda: llm_chain.predict(**full_prompt_args),
-                [openai.error.RateLimitError],
+                [
+                    openai.error.RateLimitError,
```

Comment:
tbh I think a cleaner interface might be to define a simple error dataclass that contains the exception and an optional callable e.g.
```
class ErrorWithCheck:
   exception: Exception
   check_fn: Optional[Callable[...]] = None
```

This way you avoid specifying a Union of different data types here, and also prevents blowing up the typing of the function signature for these arguments